### PR TITLE
Seperate reagents into the mobtypes they affect with on_life

### PR DESCRIPTION
**Pull request by ZomgPonies**
- Makes IPCs immune to pretty much all organic compounds except a few adminbooze ones.
- Makes IPCs immune to MOST but not all inorganic compounds (uranium and plasma for now, discuss)
- This has no impact on getting splashed with shit, only injecting/swallowing/eating it.

### CODE REVIEW

File: code/modules/mob/living/carbon/human/life.dm [hunk 1]
```diff
@@ -722,7 +722,10 @@ var/global/list/brutefireloss_overlays = list("1" = image("icon" = 'icons/mob/sc
 
 	proc/handle_chemicals_in_body()
 
-		if(reagents) //Synths don't process reagents. // fuck you they do now - Iamgoofball
+		if(reagents)
+			var/datum/reagent/R = reagents
```
Comment (Iamgoofball) [inline code/modules/mob/living/carbon/human/life.dm]: wait what

you're setting the reagents holder as a reagent

this is going to runtime

this needs to be:
for(var/datum/reagent/R in reagents.reagent_list)
if(!R.affected & SYNTHETIC && species.flags & IS_SYNTHETIC) return

**Comment by tigercat2000** [comment]
Going to suggest some potential points to discuss
 - Pills 
  - PR does not currently prevent IPC's from being fed pills
 - Uranium and Plasma affecting IPC's
  - Only applies to reagent versions
  - Note that Plasma causes toxin damage which will be unrepairable due to immunity of [insert anti-toxin name here]

**Comment by Fox-McCloud** [comment]
Wouldn't it be easier to just have reagents not impact synthetics, by default, but check for a flag of the nature of "IMPACTS_SYNTHETICS"  or something?

It would be easier to tag the reagents that would impact them than to define ORGANIC for every single reagent in the entire game (there's way more reagents than just in the "reagents" file).

**Comment by ZomgPonies** [comment]
That's exactly what this does, read it carefully.

**Comment by Fox-McCloud** [comment]
Then why the need for the ORGANIC flag?

**Comment by ZomgPonies** [comment]
In case we ever add reagents that DON'T affect organics.

**Comment by Fox-McCloud** [comment]
Ehhhh. Going to ugly+messy to define ORGANIC for 99% of the chems.

Wouldn't it be better to just check for "!SYNTHETIC"->RETURN in those very rare cases?

**Comment by ZomgPonies** [comment]
Too late now, I've already started it :/

I sort of wish our reagents were sub-divided better into medicinals and such instead of just files but meh.

**Comment by ZomgPonies** [comment]
Ah fuckit, I'll fix it up tomorrow your way.

**Comment by Fox-McCloud** [comment]
Went through the list of reagents--these are most 'logical':

Full Damage+Effects
-hellwater
-Holywater
-Chlorine
-Thermite
-Frostoil
-Ectoplasm
-CLF3
-Phlogiston
-Napalm
-Cryostylane
-Pyrosium
-Ants

Partial Damage+Effects
-Flourine (burn)
-Sulfuric Acid (burn)
-Fluorosulfuric Acid (burn)

prevented touch reactions
-Silver Sulfa
-Styptic
-Synth Flesh
-Strange reagent
-Histamine
-Hairgrow
-Super Hairgrow
-Hair Dye
-Fliptonium
-Atrazine

**Comment by Iamgoofball** [comment]
might wanna throw cryostal in there too

**Comment by veganzombeh** [comment]
The only potential problem I see with this is that if an organic consumes a synthetic-only chemical, they won't metabolize it. It will just sit there in their reagents until it's somehow removed.

**Comment by ZomgPonies** [comment]
I did this upside down, it should be redone from scratch @Fox-McCloud

**Pull request closed without merging**